File: Documentation/5.4/Samples/python/ClientApi/Operations/Maintenance/Indexes/GetIndexTerms.py
from typing import List, Optional

from ravendb import GetTermsOperation
from ravendb.documents.operations.definitions import MaintenanceOperation

from examples_base import ExampleBase


class GetIndexTerms(ExampleBase):
    def setUp(self):
        super().setUp()

    def test_get_index_terms(self):
        with self.embedded_server.get_document_store("GetIndexTerms") as store:
            # region get_index_terms
            # Define the get terms operation
            # Pass the requested index-name, index-filed, start value & page size
            get_terms_op = GetTermsOperation("Orders/Totals", "Employee", "employees/5-A", 10)

            # Execute the operation by passing it to maintenance.send
            field_terms = store.maintenance.send(get_terms_op)

            # field_terms will contain alle the terms that come after term 'employees/5-A' for index-field 'Employee'
            # endregion

    class Foo:
        # region get_index_terms_syntax
        class GetTermsOperation(MaintenanceOperation[List[str]]):
            def __init__(self, index_name: str, field: str, from_value: Optional[str], page_size: int = None): ...

        # endregion

File: Documentation/5.4/Samples/php/ClientApi/Operations/Maintenance/Indexes/GetIndexTerms.php
<?php

namespace RavenDB\Samples\ClientApi\Operations\Maintenance\Indexes;

use RavenDB\Documents\DocumentStore;
use RavenDB\Documents\Operations\Indexes\GetTermsOperation;
use RavenDB\Type\StringArrayResult;

class GetIndexTerms
{
    public function samples(): void
    {
        $store = new DocumentStore();
        try {
            # region get_index_terms
            // Define the get terms operation
            // Pass the requested index-name, index-field, start value & page size
            $getTermsOp = new GetTermsOperation("Orders/Totals", "Employee", "employees/5-a", 10);

            // Execute the operation by passing it to Maintenance.Send
            /** @var StringArrayResult $fieldTerms */
            $fieldTerms = $store->maintenance()->send($getTermsOp);

            // fieldTerms will contain the all terms that come after term 'employees/5-a' for index-field 'Employee'
            # endregion
        } finally {
            $store->close();
        }
    }
}

/*
interface IFoo
{
    # region get_index_terms_syntax
    GetTermsOperation(?string $indexName, ?string $field, ?string $fromValue, ?int $pageSize = null)
    # endregion
}
*/
